src/gigs-board/components/posts/List.jsx

```diff
@@ -148,6 +150,76 @@ if (props.searchResult) {
   postIds = Near.view(nearDevGovGigsContractAccountId, "get_children_ids");
 }
 
+const ONE_DAY = 60 * 60 * 24 * 1000;
+const ONE_WEEK = 60 * 60 * 24 * 1000 * 7;
+const ONE_MONTH = 60 * 60 * 24 * 1000 * 30;
+
+function getHotnessScore(post) {
+  //post.id - shows the age of the post, should grow exponentially, since newer posts are more important
+  //post.likes.length - linear value
+  const age = Math.pow(post.id, 5);
+  const comments = post.comments;
+  const commentAge = comments.reduce((sum, age) => sum + Math.pow(age, 5), 0);
+  const totalAge = age + commentAge;
+  //use log functions to make likes score and exponentially big age score close to each other
+  return Math.log10(post.likes.length) + Math.log(Math.log10(totalAge));
+}
+
+const getPeriodText = (period) => {
+  let text = "Last 24 hours";
+  if (period === "week") {
+    text = "Last week";
+  }
+  if (period === "month") {
+    text = "Last month";
+  }
+  return text;
+};
+
+const findHottestsPosts = (postIds, period) => {
+  let allPosts;
+  if (!state.allPosts) {
+    allPosts = Near.view("devgovgigs.near", "get_posts");
+    if (!allPosts) {
+      return [];
+    }
+    State.update({ allPosts });
+  } else {
+    allPosts = state.allPosts;
+  }
+  let postIdsSet = new Set(postIds);
+  let posts = allPosts.filter((post) => postIdsSet.has(post.id));
+
+  let periodTime = ONE_DAY;
+  if (period === "week") {
+    periodTime = ONE_WEEK;
+  }
+  if (period === "month") {
+    periodTime = ONE_MONTH;
+  }
+  const periodLimitedPosts = posts.filter((post) => {
+    const timestamp = post.snapshot.timestamp / 1000000;
+    return Date.now() - timestamp < periodTime;
+  });
+  const modifiedPosts = periodLimitedPosts.map((post) => {
+    const comments =
+      Near.view("devgovgigs.near", "get_children_ids", {
+        post_id: post.id,
+      }) || [];
+    post = { ...post, comments };
+    return {
+      ...post,
+      postScore: getHotnessScore(post),
+    };
+  });
+  modifiedPosts.sort((a, b) => b.postScore - a.postScore);
+  return modifiedPosts.map((post) => post.id);
+};
+
+if (props.recency == "hot") {
+  postIds = findHottestsPosts(postIds, state.period);
+}
```

Comment:
Move here the conditions that we have above, that pull `postIds`, so we don't [fetch top-level posts](https://github.com/near/devgigsboard-widgets/pull/94/files#diff-39121cdc76ae3bae9ebc061f56ba14eb047d13a60474dc83e83edf13ce55d314R150) unnecessarily.

Reply:
This can't be simply moved above. `findHottestsPosts` process on results of filtering/searching so it's a condition checked on all cases.

Reply:
I tried a few of ways to rewrite this if logic, but not able to figure out a better alternative. They are both more repetitive in combining the conditions and introduce more level of nestings that's less readable. Do you have a better way to combine filter, search and sorting (at frontend)? Eventually I'm lean for this approach: https://www.loom.com/share/77ab9a4acbf34e1aa65961dc70459a22l, that will also resolves the performance problem.